### PR TITLE
Completion fixes

### PR DESCRIPTION
**Pull request by rcorre**
Fix two recent issues and remove one old TODO.

---
This change is [Reviewable](https://reviewable.io/reviews/qutebrowser/qutebrowser/2846)

### CODE REVIEW

File: qutebrowser/completion/models/histcategory.py [hunk 1]
```diff
@@ -90,13 +90,8 @@ def set_pattern(self, pattern):
...
         self.setQuery(self._query)
 
-    def delete_cur_item(self, index):
-        """Delete the row at the given index."""
-        if not self.delete_func:
-            raise cmdexc.CommandError("Cannot delete this item.")
-        data = [self.data(index.sibling(index.row(), i))
-                for i in range(self.columnCount())]
-        self.delete_func(data)
+    def removeRow(self, _row, _col):
```
Comment (The-Compiler) [inline qutebrowser/completion/models/histcategory.py]: There's [QAbstractItemModel::removeRow](http://doc.qt.io/qt-5/qabstractitemmodel.html#removeRow) already, which mentions that it calls `removeRows`. Also, `removeRow` is not `virtual` in C++, so if this gets called by Qt internally for some reason, it wouldn't use the overloaded version - so I think it'd be better to implement `removeRows` and then use `removeRow` without overloading that.

**Comment by rcorre** [comment]
Review status: 0 of 8 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[qutebrowser/completion/models/histcategory.py, line 93 at r1](https://reviewable.io:443/reviews/qutebrowser/qutebrowser/2846#-KpgNFydrFd8qpjyIxyg-r1-93:-KplLvTHwZClz9Diov-I:b-896fix) ([raw file](https://github.com/qutebrowser/qutebrowser/blob/00be9e3c7fb269aa26942f6fcaf374c9b88cc8f6/qutebrowser/completion/models/histcategory.py#L93)):*
<details><summary><i>Previously, The-Compiler (Florian Bruhin) wrote…</i></summary><blockquote>

There's [QAbstractItemModel::removeRow](http://doc.qt.io/qt-5/qabstractitemmodel.html#removeRow) already, which mentions that it calls `removeRows`. Also, `removeRow` is not `virtual` in C++, so if this gets called by Qt internally for some reason, it wouldn't use the overloaded version - so I think it'd be better to implement `removeRows` and then use `removeRow` without overloading that.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/qutebrowser/qutebrowser/2846)*

**Comment by rcorre** [comment]
@The-Compiler I think you just have to acknowledge one more comment

**Comment by The-Compiler** [comment]
I'd prefer not using reviewable.io for small simple PRs, so I just reviewed it on GitHub - time to merge now :wink:

**Pull request merged**